src/utils/authFetch.js

```diff
@@ -2,19 +2,28 @@ import {jwtUtils} from "./jwt/jwtUtils";
 import {loginAPI} from "./apis/LoginAPI";
 import {checkedFetch} from "./chekedFetch";
 
-export const authFetch = (url, data, error) => {
+export const authFetch = (url, data, errorMessage) => {
     const authToken = jwtUtils.getAuthToken()
     const dataHeaders = data?.headers
     const authHeaders = authToken ? {[loginAPI.jwtTokenKey]: authToken} : {}
     const newHeaders = dataHeaders ? Object.assign(dataHeaders, authHeaders) : authHeaders
     const authData = Object.assign(data || {}, {headers: newHeaders})
-    return checkedFetch(url, authData, error)
-        .catch(() => jwtUtils.refreshToken().then(() => {
-            authData.headers[loginAPI.jwtTokenKey] = jwtUtils.getAuthToken()
-            return checkedFetch(url, authData).catch(() => {
-                localStorage.removeItem(loginAPI.refreshTokenStorageKey)
-                localStorage.removeItem(loginAPI.authTokenStorageKey)
-                window.location.reload()
-            })
-        }))
+    return checkedFetch(url, authData, errorMessage)
+        .catch(httpError => {
+            if(httpError.code && httpError.code === 401)
+                jwtUtils.refreshToken().then(() => {
+                    authData.headers[loginAPI.jwtTokenKey] = jwtUtils.getAuthToken()
+                    return checkedFetch(url, authData, errorMessage).catch(httpError => {
+                        if(httpError.code && httpError.code === 401) {
```

Comment:
Czemu drugi raz?

Reply:
Racja, o jeden skok za daleko

Reply:
A nie, jednak jest git. To jest obsługa sytuacji, kiedy:
- uda się odświeżyć token -> propagujemy błąd
- nie uda się odświeżyć -> musimy zrobić wylogowanie